app/models/user.rb

```diff
@@ -35,6 +38,7 @@ def self.find_for_database_authentication(warden_conditions)
   def self.from_omniauth(auth)
     where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
       user.email = auth.info.email
+      user.username = "#{auth.info.email.split('@').first}_#{rand(1..9).to_s}"
```

Comment:
Don't use random as it may not solve problems of users with similar usernames. 
Look to see if Postgres has a solution for this. 🤓